test/functional/test_framework/test_shell.py

```diff
@@ -7,6 +7,14 @@
 from test_framework.test_framework import BitcoinTestFramework
 
 
+# BitcoinTestFramework instances are supposed to be constructed with the path
+# of the calling test in order to find shared data like configuration and the
+# cache. Since TestShell is meant for interactive use, there is no concrete
+# test; passing a dummy name is fine though, as only the containing directory
+# is relevant for successful initialization.
+tests_directory = pathlib.Path(__file__).absolute().parent.parent
+dummy_testshell_file = tests_directory / "testshell_dummy.py"
```

Comment:
Q: I'm not sure I understand the reason for extraction, can you please clarify that?

Nit: We're adding another global here which pollutes the namespace, can we rather inline it?
```suggestion
dummy_testshell_file = pathlib.Path(__file__).absolute().parent.parent / "testshell_dummy.py"
```

Reply:
The reason for the extraction was to share the variable with the `reset` function, I have however reverted the extraction to avoid using global variables